ref/general/intro-dev-mode.adoc

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2020 IBM Corporation and others.
+// Licensed under Creative Commons Attribution-NoDerivatives
+// 4.0 International (CC BY-ND 4.0)
+//   https://creativecommons.org/licenses/by-nd/4.0/
+//
+// Contributors:
+//     IBM Corporation
+//
+:page-description: When you run Open Liberty in development mode, you can rapidly code, deploy, test, and debug applications directly in your IDE or text editor.
+:seo-title: Development mode
+:seo-description: When you run Open Liberty in development mode, you can rapidly code, deploy, test, and debug applications directly in your IDE or text editor.
+:page-layout: general-reference
+:page-type: general
+= Introduction to development mode
+
+When you run Open Liberty in development mode (dev mode), you can rapidly code, deploy, test, and debug applications directly in your internal development environment (IDE) or text editor.
+
+With dev mode, you can quickly iterate on changes to your code and get immediate feedback from on-demand or automatic unit and integration tests. You can also attach a debugger to step through your code at any time.
+Dev mode reduces turnaround times and manages the details of testing and debugging so you can focus on your code.
+You don't need to rebuild your application to implement changes, run tests, or debug.
+Dev mode supports a streamlined workflow for iterative application development.
+
+== What is dev mode?
+
+Dev mode is a goal of link:https://github.com/OpenLiberty/ci.maven[the Open Liberty Maven plug-in].
+It integrates a set of capabilities for Open Liberty so you can edit and monitor your application in real time, without restarting your running server.
+Dev mode provides an enhanced developer experience in three primary focus areas: deploying changes, running tests, and debugging.
+
+To start dev mode, <<run, install the Open Liberty Maven plug-in>> and run the `mvn liberty:dev` command.
+
+=== Detect, recompile, and pick up code changes
+
+In dev mode, your running server can automatically detect, recompile, and deploy code changes whenever you save a new change in your IDE or text editor.
+
+Dev mode automatically detects the following changes to your application source:
+
+- Java source file and test file changes
+- Resource file changes
+- Configuration directory and configuration file changes
+- Addition of new dependencies to your pom.xml file
+- Addition of new features in the Open Liberty server configuration
+
+Resource file, configuration file, and configuration directory changes are copied into your target directory.
+New dependencies in your `pom.xml` file are added to your class path.
+New features are installed and started.
+
+Dev mode does not detect some changes, such as changes to the host and port for Open Liberty or adding a configuration directory or file.
+To detect these changes, you must restart dev mode.
+To restart, first exit dev mode by typing `Control-C`, or by typing `q` and pressing `Enter`.
```

Comment:
The guides use `CTRL+C` rather than `Control-C`. I doubt either is incorrect, but we could use what the guides do for consistency across documentation.